Packs/QualysFIM/Integrations/Qualys/QualysFIM.py

```diff
@@ -0,0 +1,643 @@
+                    break
+            item_dictionary = remove_empty_elements({'Hostname': hostname,
+                                                     'Last Activity': last_checked_in,
+                                                     'Creation Time': created,
+                                                     'Agent Version': agent_version,
+                                                     'Driver Version': driver_version,
+                                                     'Last Agent Update': agent_last_updated,
+                                                     'Asset ID': asset_id})
+            outputs.append(item_dictionary)
+    readable_output = tableToMarkdown(name=f'Listed {len(outputs)} Assets:', t=outputs,
+                                      headers=table_headers, removeNull=True)
+    return CommandResults(outputs_prefix='QualysFIM.Asset',
+                          outputs_key_field='id',
+                          raw_response=raw_response,
+                          outputs=raw_outputs,
+                          readable_output=readable_output)
+
+
+def fetch_incidents(client: Client, last_run: Dict[str, int],
+                    max_fetch: str,
+                    first_fetch_time: str) -> Tuple[Dict[str, int], List[dict]]:
+    """
+    Fetch incidents (alerts) each minute (by default).
+    Args:
+        client (Client): Sophos Central Client.
+        last_run (dict): Dict with last_fetch object,
+                                  saving the last fetch time(in millisecond timestamp).
+        first_fetch_time (dict): Dict with first fetch time in str (ex: 3 days ago).
+        max_fetch (str): Max number of alerts to fetch.
+    Returns:
+        Tuple of next_run (millisecond timestamp) and the incidents list
+    """
+    last_fetch_timestamp = last_run.get('last_fetch', None)
+    if last_fetch_timestamp:
+        last_fetch_date = datetime.fromtimestamp(last_fetch_timestamp)
+        last_fetch = last_fetch_date
+    else:
+        first_fetch_date = dateparser.parse(first_fetch_time).replace(tzinfo=None)
+        last_fetch = first_fetch_date
+
+    next_run = last_fetch
+    last_fetch_timestamp_new = int(datetime.timestamp(next_run) * 1000)
+    time_now = int(datetime.timestamp(datetime.now()) * 1000)
+
+    if int(max_fetch) > 200:
+        raise ValueError('Max Fetch is limited to 200 incidents per fetch, '
+                         'please choose lower number than 200')
+
+    params = {'pageSize': max_fetch,
+              'filter': f"createdBy.date: ['{last_fetch_timestamp_new}'..'{time_now}']",
+              'sort': '[{"createdBy.date":"asc"}]'}
+    raw_response = client.incidents_list(params)
+    incidents = []
+    for incident in raw_response:
+        incident = incident.get('data', None)
+        incident_id = incident.get('id', None)
+        incident_name = incident.get('name', None)
+        created_date = dict_safe_get(incident, ['createdBy', 'date'])
```

Comment:
Is there a possibility that this will not get the date value?

Reply:
no the response will always contain the creation date of the incident